### ISSUE TITLE
styling: Fix `auto` inheriting "old" attributes

### ISSUE THREAD

**Issue by johannes-wolf** [johannes-wolf/cetz#143]
The following example draws a red line with an red arrow (mark):
```
set-style(stroke: red)
line((0,0), (1,0), mark: (end: ">"))
```

This example on the other hand, draws a blue line with a (still) red arrow head:
```
set-style(stroke: red)
line((0,0), (1,0), mark: (end: ">"), stroke: blue)
```

This is, because the `auto` stroke attribute of `mark` gets set to the root `stroke` of the context.
But I expect it to get set to the attribute set as style (blue). @fenjalien Any thoughts on this?

**Comment by fenjalien** [comment]
In a way this is expected behavior as that mark style inherits from the root style.
It could be changed so that styles inherit directly but that would cause a similar issue in the other direction. you would need to specify the stroke of mark if you want it different to the stroke of the line.

I think the best way would be to allow a style value that causes direct inheritance instead of from the root.
- `auto` inherits from the root style
- `"inherit"` inherits from the styles direct parent.

**Comment by johannes-wolf** [comment]
In which case do you expect some style not to inherit the "root" style instead of the one just set, though?

**Comment by fenjalien** [comment]
The style should inherit as much as it could from its parent, if it does not exist on the parent it should inherit from the style's root.

e.g.
```
(line: (mark: "inherit", ...))
```
`mark` should inherit as much as it can (stroke, fill) but the rest of the mark styling for line should take on the root style of mark (arrows...).
```
(line: (mark: (stroke: "inherit")))
```
the mark's stroke should inherit from line but all other mark styling should be the root mark style (fill..)

I think the best way to look at it is which direction we traverse the style dictionary to find the first instance of what to inherit from. `auto` is into the dictionary from the root, `"inherit"` is up from the current position.

**Comment by johannes-wolf** [comment]
With the current implementation `auto` only inherits form the very "bottom" though.
But the differentiation between `auto` and `inherit` makes no sense to me.
When do you want this:
```
set-style(stroke: red)
line(.., stroke: blue, mark: (end: ">"))
```
have red marks? Therefore I propose to always inherit top-down instead of bottom-up, like in CSS.

**Comment by johannes-wolf** [comment]
Link #165

**Comment by johannes-wolf** [comment]
Fixed with #200.
